pkg/storegateway/indexheader/fileutil/mmap.go

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+// Provenance-includes-location: https://github.com/prometheus/prometheus/blob/main/tsdb/fileutil/mmap.go
+// Provenance-includes-license: Apache-2.0
+// Provenance-includes-copyright: The Prometheus Authors.
+
+package fileutil
+
+import (
+	"os"
+
+	"github.com/pkg/errors"
+)
+
+type MmapFile struct {
+	f *os.File
+	b []byte
+}
+
+func OpenMmapFile(path string, populate bool) (*MmapFile, error) {
```

Comment:
as far as I understand this file is basically copied from Prometheus and then modified, right? 
would it be worth it to upstream this change into Prometheus?

Reply:
or if we don't want to upstream into OSS prometheus, we could still consider making the change in `mimir-prometheus`.

Reply:
Given the size of the files, I don't think it's worth the effort doing either right now.

If the testing goes well, then I'll upstream the changes.